packages/modular-scripts/src/utils/filterDependencies.ts

```diff
@@ -18,22 +19,54 @@ export function filterDependencies(
     };
   }
 
-  // By default, nothing is in blocklist
   const externalBlockList =
     process.env.EXTERNAL_BLOCK_LIST && !isApp
       ? process.env.EXTERNAL_BLOCK_LIST.split(',')
-      : [];
+      : undefined;
 
-  // By default, everything in allow list
   const externalAllowList =
     process.env.EXTERNAL_ALLOW_LIST && !isApp
       ? process.env.EXTERNAL_ALLOW_LIST.split(',')
-      : ['*'];
+      : undefined;
+
+  logger.debug('Filtering dependencies...');
+  logger.debug(
+    `External block list for dependencies is: ${JSON.stringify(
+      externalBlockList,
+    )}`,
+  );
+  logger.debug(
+    `External allow list for dependencies is: ${JSON.stringify(
+      externalAllowList,
+    )}`,
+  );
 
+  return matchDependencies({
+    packageDependencies,
+    allowList: externalAllowList,
+    blockList: externalBlockList,
+  });
+}
+
+export function matchDependencies({
+  packageDependencies,
+  // By default, everything in allow list
+  allowList = ['**'],
```

Comment:
Re-reading this, I think "allow", "block", and even "matchDependencies" (and "filterDependencies") are not self-descriptive. Really what you're looking for is external vs bundled. It'll be very easy to forget what this function is trying to do or what it's for.

Reply:
@benpryke I changed terminology to reflect extarnal / bundled in the function that was introduced in this PR, and renamed it to `partitionDependencies`.
I didn't change it in the rest of the file, since `EXTERNAL_BLOCK_LIST` and `EXTERNAL_ALLOW_LIST` were approved months ago, are part of the user interface and not in scope of this PR.